src/System.Net.Sockets.Legacy/tests/FunctionalTests/DnsEndPointTest.cs

```diff
@@ -58,8 +60,8 @@ public void Socket_ReceiveFromDnsEndPoint_ArgumentException()
         {
             using (Socket sock = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
             {
-                sock.Bind(new IPEndPoint(IPAddress.Loopback, 8080));
-                EndPoint endpoint = new DnsEndPoint("localhost", 8080);
+                sock.Bind(new IPEndPoint(IPAddress.Loopback, TestPortBase + 4));
```

Comment:
Are these ```+ 3```, ```+ 4```, etc. being used in order to have different ports used for each test?  If so, that seems fragile and easy to get wrong, resulting in sporadic test failures.  Could we instead have a field or something that tracked the next available port, and do an Interlocked.Increment on it to get the ID for the test, or something like that?